### PR TITLE
Quality reinforcement (3): fix unstable test cases

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
fix unstable test cases in CI.

- [x] testScheduler.TestScheduler, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6272/pipeline/70), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/8ef10bbf4207e3c5a29fdbfd40a5ca23c00fb016
- [x] ha: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6432/pipeline) [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6459/pipeline/71/), lance6716, https://github.com/pingcap/dm/pull/1007/commits/db840d9c1d80b57c5f776958636ecec0d496034b
   **root cause: leader client is not reflecting real time leader change, so there maybe two DM-master thinking the other is leader and cause forwarding loop**
- [x] testSubTask.TestSubTaskNormalUsage, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6546/pipeline), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/29299b1d27e220fbbb277390b7b4d9254fad6ee9
- [x] testMaster.TestOfflineMember, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6577/pipeline), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/efc666396f737970900fe0c6d20af903e19b4c53, https://github.com/pingcap/dm/pull/1007/commits/ae03550a6a8e49258c99bab46da1f731982fc422
- [ ] shardddl3 DM-103, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6661/pipeline)
- [x] testServer.TestWatchSourceBoundEtcdCompact, testWorkerEtcdCompact.TestWatchRelayStageEtcdCompact
testWorkerEtcdCompact.TestWatchSubtaskStageEtcdCompact :`bind: address already in use`: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6674/pipeline):  csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/924683d1d1ec93c76465c8b81f408d3d5400dc83
- [x] testScheduler.TestScheduler: `sources [mysql-replica-1] have not bound`: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6674/pipeline), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/b488f5f793a7e48a6141acf9fe3f91701d3e343a
- [x] testMaster.TestFailToStartLeader: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6674/pipeline), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/87e940e21362b43fd401e1b5152f573f4d864a66
- [x] dmctl_basic, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6664/pipeline/), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/4e0070b167b7679b866afb633172beb4056a7499
- [x] testPessimist.TestSourceReEntrant, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6683/pipeline/71), csuzhangxc, https://github.com/pingcap/dm/pull/1007/commits/cb98757ada8ab3e270b368b79917a9b92acff586
  the put op (`2020/09/10 16:24:56.189 +08:00`) is too slow after receive info (`2020/09/10 16:24:54.167 +08:00`)

### What is changed and how it works?
- /run-all-tests to find unstable cases.
- push commit in this pr to fix the cases.

### CODE REVIEW

**Review by GMHDBJD — COMMENTED** [review]
LGTM

File: dm/master/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -410,6 +420,11 @@ func (t *testScheduler) testSchedulerProgress(c *C, restart int) {
 	rebuildScheduler(ctx)
 
 	// CASE 4.6: try remove source when subtasks exist.
+	// wait for source bound recovered
+	c.Assert(utils.WaitSomething(30, 100*time.Millisecond, func() bool {
+		bounds := s.BoundSources()
+		return len(bounds) == 2 && bounds[0] == sourceID1 && bounds[1] == sourceID2
```
Comment (lance6716) [inline dm/master/scheduler/scheduler_test.go:426]: will it has another bound order? @csuzhangxc

Comment (csuzhangxc) [reply dm/master/scheduler/scheduler_test.go:426]: no, we sorted in `BoundSources`.

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**